rest/src/main/java/fi/otavanopisto/pyramus/rest/CommonRESTService.java

```diff
@@ -355,6 +355,17 @@ public Response createSubject(fi.otavanopisto.pyramus.rest.model.Subject entity)
     return Response.ok().entity(objectFactory.createModel(commonController.createSubject(code, name, educationType))).build();
   }
   
+  @Path("/subjectByCode/{CODE}")
+  @GET
+  @RESTPermit (CommonPermissions.FIND_SUBJECT)
+  public Response findSubjectByCode(@PathParam("CODE") String code) {
+    Subject subject = commonController.findSubjectByCode(code);
```

Comment:
Is this going to be case-sensitive? Not sure, just wondering if that can pose an issue.

Reply:
Very much case-sensitive. That's why subjectDAO.findByCode has extra case comparison logic.